client/services/PlanningStoreService.js

```diff
@@ -0,0 +1,149 @@
+        if (isNil(self.store)) {
+            return this.createStore()
+        }
+
+        return Promise.resolve(self.store)
+    }
+
+    this.createStore = function() {
+        const _notify = {
+            pop: () => $timeout(() => notify.pop()),
+            success: (msg) => $timeout(() => notify.success(msg)),
+            error: (msg) => $timeout(() => notify.error(msg)),
+        }
+
+        return $q.all({
+            voc: vocabularies.getAllActiveVocabularies(),
+            ingest: api('ingest_providers').query({
+                max_result: 200,
+                page: 1,
+            }),
+            privileges: privileges.loaded,
+            metadata: metadata.initialize(),
+            users: userList.getAll(),
+            desks: desks.initialize(),
+            formsProfile: api('planning_types').query({
+                max_results: 200,
+                page: 1,
+            }),
+        }).then((data) => {
+            const initialState = {
+                config,
+                deployConfig: deployConfig.config,
+                vocabularies: zipObject(
+                    data.voc._items.map((cv) => cv._id),
+                    data.voc._items.map((cv) => cv.items)
+                ),
+                ingest: {
+                    providers: data.ingest._items.map((provider) => ({
+                        name: provider.name,
+                        id: provider._id,
+                    })),
+                },
+                privileges: data.privileges,
+                subjects: metadata.values.subjectcodes,
+                genres: metadata.values.genre,
+                users: data.users,
+                desks: desks.desks._items,
+                session: {
+                    sessionId: session.sessionId,
+                    identity: session.identity,
+                },
+                urgency: {
+                    urgency: metadata.values.urgency,
+                    label: gettextCatalog.getString('Urgency'),
+                },
+                formsProfile: {},
+            }
+
+            data.formsProfile._items.forEach((p) => {
+                initialState.formsProfile[p.name] = p
+            })
+
+            // create the application store
+            self.store = createStore({
+                initialState,
+                extraArguments: {
+                    api,
+                    $location,
+                    $timeout,
+                    vocabularies,
+                    superdesk,
+                    upload,
+                    notify: _notify,
+                    privileges,
+                    notifyConnectionService,
+                    userList,
+                    desks,
+                    metadata,
+                    session,
+                    deployConfig,
+                    gettextCatalog,
+                },
+            })
+
+            $rootScope.$on('vocabularies:updated', angular.bind(this, this._reloadVocabularies))
```

Comment:
imo this shouldn't be part of createStore otherwise you register this every time you call that, instead of just once in constructor